runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7642,6 +7643,17 @@ done:;
 				goto retry;
 			}
 		}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		else {
+			if ((NULL != castClass) && J9_IS_J9CLASS_VALUETYPE(castClass)) {
```

Comment:
This should be nested ifs that only test castClass once.

Reply:
@gacholio I've rather restructured the ordering of the conditionals so that castClass' is only checked once for null, please take a look at it again